src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -728,6 +728,17 @@ default List<StudentParticipation> findByExamIdWithSubmissionRelevantResult(Long
         return filterParticipationsWithRelevantResults(participations, true);
     }
 
+    /**
+     * Get all participations belonging to a student in a course with relevant results.
+     *
+     * @param courseId the id of the course
+     * @return an unmodifiable list of participations belonging to course
+     */
+    default List<StudentParticipation> findByCourseIdAndStudentIdWithRelevantResult(Long courseId, Long studentId) {
+        List<StudentParticipation> participations = findByCourseIdAndStudentIdWithEagerRatedResults(courseId, studentId);
+        return filterParticipationsWithRelevantResults(participations, false);
```

Comment:
Can we do this filter directly in the query? Now we load too many participations and throw them away later.

Reply:
I think this normally only returns 1 participation and rather the results are filtered. So it's not a big deal in my opinion